BlazorBarcodeScanner.ZXing.JS/JsInteropClass.cs

```diff
@@ -1,73 +1,84 @@
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.startDecoding", videoElementId, width, height);
-        }
-        public static void StopDecoding(IJSRuntime jSRuntime)
-        {
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.stopDecoding");
-        }
-        public static void SetVideoInputDevice(IJSRuntime jSRuntime, string deviceId) {
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.setSelectedDeviceId", deviceId);
-        }
-        public static void SetTorchOn(IJSRuntime jSRuntime)
-        {
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.setTorchOn");
-        }
-        public static void SetTorchOff(IJSRuntime jSRuntime)
-        {
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.setTorchOff");
-        }
-        public static void ToggleTorch(IJSRuntime jSRuntime)
-        {
-            jSRuntime.InvokeVoidAsync("BlazorBarcodeScanner.toggleTorch");
-        }
-
-        [JSInvokable]
-        public static void ReceiveBarcode(string barcodeText) {
-            if (!String.IsNullOrEmpty(barcodeText)) {
-                BarcodeReceivedEventArgs args = new BarcodeReceivedEventArgs();
-                args.BarcodeText = barcodeText;
-                args.TimeReceived = DateTime.Now;
-                OnBarcodeReceived(args);
-            }
-            
-        }
-        protected static void OnBarcodeReceived( BarcodeReceivedEventArgs args) {
-            BarcodeReceivedEventHandler handler = BarcodeReceived;
-            BarcodeReceived?.Invoke(args); //same as below
-            // if(handler != null ){
-            //    handler(this, e);
-            // }
-        }
-        public static event BarcodeReceivedEventHandler BarcodeReceived;
-    }
-    public class BarcodeReceivedEventArgs : EventArgs { 
-        public string BarcodeText { get; set; }
-        public DateTime TimeReceived { get; set; } = new DateTime();
-    }
-    public delegate void BarcodeReceivedEventHandler(BarcodeReceivedEventArgs args);
-    public class VideoInputDevice
-    {
-        public string DeviceId { get; set; }
-        public string GroupId { get; set; }
-        public string Kind { get; set; }
-        public string Label { get; set; }
-
-    }
-}
+using Microsoft.JSInterop;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace BlazorBarcodeScanner.ZXing.JS
+{
+    public class JsInteropClass
+    {
+        public static ValueTask<List<VideoInputDevice>> GetVideoInputDevices(IJSRuntime jsRuntime, string message)
+        {
+            // Implemented in BlazorBarcodeScannerJsInterop.js
+
+            return jsRuntime.InvokeAsync<List<VideoInputDevice>>(
+                "BlazorBarcodeScanner.listVideoInputDevices",
+                message);
+        }
+
+        public static void StartDecoding(IJSRuntime jSRuntime, string videoElementId, int width, int height) {
+            SetVideoResolution(jSRuntime, width, height);

```

Comment:
I think we should move this out of JSInterop, either in Blazor component or into JS.

Reply:
I wanted to keep the interface that passes the actual interop layer (i.e. the JS methods called) as lean as possible. Hence I decided against having an extra method in the JS part.

However, on the other hand according to the single responsibility principle, I see little value in distributing the same functionality (`StartDecoding`) among multiple classes, as `StartDecoding(IJSRuntime jSRuntime, string videoElementId, int width, int height)` is just decorating (overloading) `StartDecoding(IJSRuntime jSRuntime, string videoElementId)`. 

At the end of the day, in my eyes, the overloading method is sort of "syntactical sugar" for the client of `JsInteropClass`.

Reply:
Ah, now I got your intention.
Looks like you are working on another branch too.
Can you pls finalize PR7 so we can merge.
Thanks!

Reply:
Sure. I'll combine all the changes in #7 . Originally I wanted to have the incrments a bit more atomic, but I just found that the changes will interfere later on anyway so it makes sense to me to have all at once. Sorry for the confusion :-D.